webui/react/src/pages/WorkspaceDetails/WorkspaceMembers.tsx

```diff
@@ -0,0 +1,305 @@
+import { Button, Dropdown, Menu, Select } from 'antd';
+import { FilterDropdownProps } from 'antd/lib/table/interface';
+import { RawValueType } from 'rc-select/lib/BaseSelect';
+import { LabelInValueType } from 'rc-select/lib/Select';
+import React, { useCallback, useMemo } from 'react';
+
+import InteractiveTable, { ColumnDef, InteractiveTableSettings } from 'components/InteractiveTable';
+import { getFullPaginationConfig } from 'components/Table';
+import TableFilterSearch from 'components/TableFilterSearch';
+import Avatar from 'components/UserAvatar';
+import { useStore } from 'contexts/Store';
+import useFeature from 'hooks/useFeature';
+import useModalWorkspaceRemoveMember from 'hooks/useModal/Workspace/useModalWorkspaceRemoveMember';
+import usePermissions from 'hooks/usePermissions';
+import useSettings, { UpdateSettings } from 'hooks/useSettings';
+import {
+  assignRolesToGroup,
+  assignRolesToUser,
+  removeRoleFromGroup,
+  removeRoleFromUser,
+} from 'services/api';
+import { V1Group, V1GroupDetails, V1RoleWithAssignments } from 'services/api-ts-sdk';
+import { Size } from 'shared/components/Avatar';
+import Icon from 'shared/components/Icon/Icon';
+import { alphaNumericSorter } from 'shared/utils/sort';
+import { User, UserOrGroup, Workspace } from 'types';
+import handleError from 'utils/error';
+import { getIdFromUserOrGroup, getName, isUser } from 'utils/user';
+
+import css from './WorkspaceMembers.module.scss';
+import settingsConfig, {
+  DEFAULT_COLUMN_WIDTHS,
+  WorkspaceMembersSettings,
+} from './WorkspaceMembers.settings';
+
+interface Props {
+  assignments: V1RoleWithAssignments[];
+  groupsAssignedDirectly: V1Group[];
+  pageRef: React.RefObject<HTMLElement>;
+  usersAssignedDirectly: User[];
+  workspace: Workspace;
+}
+
+interface GroupOrMemberActionDropdownProps {
+  name: string;
+  userOrGroup: UserOrGroup;
+  workspace: Workspace;
+}
+
+const GroupOrMemberActionDropdown: React.FC<GroupOrMemberActionDropdownProps> = ({
+  userOrGroup,
+  name,
+}) => {
+  const {
+    modalOpen: openWorkspaceRemoveMemberModal,
+    contextHolder: openWorkspaceRemoveMemberContextHolder,
+  } = useModalWorkspaceRemoveMember({
+    name,
+    userOrGroup: userOrGroup,
+    userOrGroupId: getIdFromUserOrGroup(userOrGroup),
+  });
+
+  const menuItems = (
+    <Menu>
+      <Menu.Item danger key="delete" onClick={() => openWorkspaceRemoveMemberModal()}>
+        Delete
```

Comment:
I think you did a good job explaining the action in the confirm modal, but I think it's easier if this was labeled Remove instead.